File: java/src/main/java/com/dreizak/miniball/model/ArrayPointSet.java
package com.dreizak.miniball.model;

/**
 * A {@link PointSet} that stores <i>n</i> <i>d</i>-dimensional points in a Java array of
 * {@code double}s.
 */
public final class ArrayPointSet implements PointSet
{
  private int d, n;
  private double[] c;

  /**
   * Creates an array-based point set to store <i>n</i> <i>d</i>-dimensional points.
   * 
   * @param d
   *          the dimensions of the ambient space
   * @param n
   *          the number of points
   */
  public ArrayPointSet(int d, int n)
  {
    this.d = d;
    this.n = n;
    this.c = new double[n * d];
  }

  @Override
  public int size()
  {
    return n;
  }

  @Override
  public int dimension()
  {
    return d;
  }

  @Override
  public double coord(int i, int j)
  {
    assert 0 <= i && i < n;
    assert 0 <= j && j < d;
    return c[i * d + j];
  }

  /**
   * Sets the <i>j</i>th Euclidean coordinate of the <i>i</i>th point to the given value.
   * 
   * @param i
   *          the number of the point, 0 ≤ i < {@code size()}
   * @param j
   *          the dimension of the coordinate of interest, 0 ≤ j ≤ {@code dimension()}
   * @param v
   *          the value to set as the <i>j</i>th Euclidean coordinate of the <i>i</i>th point
   */
  public void set(int i, int j, double v)
  {
    assert 0 <= i && i < n;
    assert 0 <= j && j < d;
    c[i * d + j] = v;
  }

  public String toString()
  {
    StringBuffer s = new StringBuffer("{");
    for (int i = 0; i < n; ++i)
    {
      s.append('[');
      for (int j = 0; j < d; ++j)
      {
        s.append(coord(i, j));
        if (j < d - 1) s.append(",");
      }
      s.append(']');
      if (i < n - 1) s.append(", ");
    }
    s.append('}');
    return s.toString();
  }
}

File: csharp/miniball/model/ArrayPointSet.cs
using System.Diagnostics;
using System.Text;

namespace SEB;

/// <summary>
/// A PointSet that stores <i>n</i> <i>d</i>-dimensional points in a Java array of s
/// </summary>
public class ArrayPointSet : PointSet
{

    int d, n;
    double[] c;

    /// <summary>
    /// Creates an array-based point set to store <i>n</i> <i>d</i>-dimensional points.   
    /// </summary>
    /// <param name="d">the dimensions of the ambient space</param>
    /// <param name="n">the number of points</param>
    public ArrayPointSet(int d, int n)
    {
        this.d = d;
        this.n = n;
        this.c = new double[n * d];
    }

    public int Size => n;

    public int Dimension => d;

    public double Coord(int i, int j)
    {
        Trace.Assert(0 <= i && i < n);
        Trace.Assert(0 <= j && j < d);
        return c[i * d + j];
    }

    /// <summary>
    /// Sets the <i>j</i>th Euclidean coordinate of the <i>i</i>th point to the given value.
    /// </summary>
    /// <param name="i">the number of the point, 0 ≤ i < {@code size()}</param>
    /// <param name="j">the dimension of the coordinate of interest, 0 ≤ j ≤ {@code dimension()}</param>
    /// <param name="v">the value to set as the <i>j</i>th Euclidean coordinate of the <i>i</i>th point</param>
    public void Set(int i, int j, double v)
    {
        Trace.Assert(0 <= i && i < n);
        Trace.Assert(0 <= j && j < d);
        c[i * d + j] = v;
    }

    public override string ToString()
    {
        var sb = new StringBuilder('{');
        for (int i = 0; i < n; ++i)
        {
            sb.Append('[');
            for (int j = 0; j < d; ++j)
            {
                sb.Append(Coord(i, j));
                if (j < d - 1) sb.Append(",");
            }
            sb.Append(']');
            if (i < n - 1) sb.Append(", ");
        }
        sb.Append('}');

        return sb.ToString();
    }

}
